index.js

```diff
@@ -1,62 +1,35 @@
 const express = require('express');
-const cors = require('cors');
-const { ApolloServer } = require('apollo-server-express');
-const jwt = require('jsonwebtoken');
+const { connectDatabase } = require('./src/database');
 const { passport } = require('./src/auth');
-const db = require('./src/database');
-
-const PORT = process.env.PORT || 5000;
-
+const { buildApolloServer } = require('./src/graphql');
 const { register: registerRoutes } = require('./src/routes');
+const { initializeCors } = require('./src/cors');
+const { generateJwt } = require('./src/token');
 
-const typeDefs = require('./src/graphql/typeDefs');
-const resolvers = require('./src/graphql/resolvers');
-
-const server = new ApolloServer({
-    typeDefs,
-    resolvers,
-    introspection: true,
-    playground: true,
-});
+const PORT = process.env.PORT || 5000;
 
 const app = express();
 
-app.use(express.static('public'));
+connectDatabase();
 
-// TODO: Use environment variable
-const whitelist = process.env.CORS_WHITELIST.split(',');
-const corsOptions = {
-    origin: (origin, callback) => {
-        if (whitelist.indexOf(origin) !== -1 || !origin) {
-            callback(null, true);
-        } else {
-            callback(new Error('Not allowed by CORS'));
-        }
-    },
-};
-
-app.use(cors(corsOptions));
+app.use(express.static('public'));
+app.use(initializeCors());
 app.use(passport.initialize());
 
 app.get('/auth/google', passport.authenticate('google-oauth-jwt', {
     callbackUrl: process.env.GOOGLE_OAUTH_CALLBACK_URL,
     scope: 'email',
 }));
-
 app.get('/auth/google/callback', passport.authenticate('google-oauth-jwt', {
     callbackUrl: process.env.GOOGLE_OAUTH_CALLBACK_URL,
 }), (req, res) => {
-    const token = jwt.sign({
-        email: 'justinyum98@gmail.com',
-    }, process.env.JWT_SECRET, {
-        expiresIn: '14d',
-    });
-    res.cookie('jwt', token);
+    res.cookie('jwt', generateJwt());
```

Comment:
`generateJwt()` creates a JWT token inside `token.js`, rather than the entry-point file.